packages/dockerApi/src/dockerComposeUp.ts

```diff
@@ -35,50 +39,76 @@ export async function dockerComposeUpPackage({
     throw Error(`No docker-compose found for ${dnpName} at ${composePath}`);
   }
 
-  // TODO: restartDappmanagerPatch should be placed here. Its not placed in this module
-  // because it adds a lot of complexity to create a new module for it.
-
-  // Add timeout option if not previously specified
-  const timeout = containersStatus ? getDockerTimeoutMax(Object.values(containersStatus)) : undefined;
-  if (timeout && dockerComposeUpOptions && !dockerComposeUpOptions.timeout) dockerComposeUpOptions.timeout = timeout;
+  // In-process lock: prevent concurrent up for the same composePath
+  if (composeUpLocks.has(composePath)) {
+    logs.warn(`dockerComposeUpPackage: composePath ${composePath} is already being processed. Skipping concurrent call.`);
+    throw new Error(`dockerComposeUpPackage: composePath ${composePath} is already being processed.`);
+  }
 
-  // Check the current status of package's container if any
-  const serviceNames: string[] = readComposeServiceNames(composePath);
-  const servicesToStart = serviceNames.filter(
-    (serviceName) => containersStatus && containersStatus[serviceName]?.targetStatus !== "stopped"
-  );
+  const upPromise = (async () => {
+    // TODO: restartDappmanagerPatch should be placed here. Its not placed in this module
+    // because it adds a lot of complexity to create a new module for it.
+
+    // Add timeout option if not previously specified
+    const timeout = containersStatus ? getDockerTimeoutMax(Object.values(containersStatus)) : undefined;
+    if (timeout && dockerComposeUpOptions && !dockerComposeUpOptions.timeout) dockerComposeUpOptions.timeout = timeout;
+
+    // Check for any service in 'removing' state before proceeding
+    const dnpData: InstalledPackageData | null = await listPackageNoThrow({ dnpName });
+    if (dnpData && dnpData.containers) {
+      const removingContainers = dnpData.containers.filter(c => c.state === "removing");
+      if (removingContainers.length > 0) {
+        const serviceNames = removingContainers.map(c => c.serviceName).join(", ");
+        logs.warn(`dockerComposeUpPackage: Cannot proceed, services in 'removing' state: ${serviceNames}`);
```

Comment:
I would remove warn log since u are already throwing and this error should be logged afterwards